### PR TITLE
Added a basic module for snap packages (packaging/os/snap.py)

### PR DESCRIPTION
**Pull request by vcarceler**
##### SUMMARY

Added a basic module for managing snap packages.

Snap is a new package format, for more info visit: https://snapcraft.io/

This module is able to install, uninstall snaps with or without --classic confinement.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

snap
packaging/os/snap.py

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/packaging/os/snap.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    snap_path = module.get_bin_path("snap", True)
+    cmd = "%s list %s" % (snap_path, name)
+    rc, stdout, stderr = module.run_command(cmd, check_rc=False)
+
+    return rc, stdout, stderr
+
+def install_snap(module, name):
+    classic = ''
+    if module.params['classic']:
+        classic = '--classic'
```
Comment (bittner) [inline lib/ansible/modules/packaging/os/snap.py:97]: I would write this all on a single line as:
```python
classic = '--classic' if module.params['classic'] else ''
```

File: lib/ansible/modules/packaging/os/snap.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+    # manipulate or modify the state as needed (this is going to be the
+    # part where your module will do what it needs to do)
+
+    # Test if snap exists
+    rc, out, err = info_snap(module, module.params['name'])
+    if module._diff:
+        diff = parse_diff(out)
+    else:
+        diff = {}
```
Comment (bittner) [inline lib/ansible/modules/packaging/os/snap.py:155]: I would write this all on a single line as:
```python
diff = parse_diff(out) if module._diff else {}
```

**Comment by bittner** [comment]
@vcarceler Could you address the complaints of the @ansibot [from above](https://github.com/ansible/ansible/pull/40852#issuecomment-392954299), please? Then this PR could move forward (into a discussion).

**Comment by angristan** [comment]
@bittner See #44939

**Comment by bittner** [comment]
Then we should close this PR?

**Comment by angristan** [comment]
I think we should! I'm motivated to finish the module

**Comment by webknjaz** [comment]
Implemented by @angristan via #44939 on top of this PR. Thanks to @vcarceler for the original work!

**Comment by angristan** [comment]
Thanks @vcarceler!

**Comment by vcarceler** [comment]
Thanks @angristan, @webknjaz!

I'm very happy :-)

**Pull request closed without merging**